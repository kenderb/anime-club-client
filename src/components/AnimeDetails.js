import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { fetchAnimeDetail, setFavoriteAnime } from '../actions';
import './AnimeDetails.style.css';

const AnimeDetails = ({
  match, animeDetails, user, setFavoriteAnime, favorites,
  fetchAnimeDetail,
}) => {
  const { id } = match.params;
  const { title, url_image: imageUrl, description } = animeDetails;

  const handleFavorite = () => {
    setFavoriteAnime({ userId: user.id, animeId: id });
  };

  useEffect(() => {
    fetchAnimeDetail(id);
  }, [fetchAnimeDetail]);

  if (!user.loggedIn) return <Redirect to="/" />;

  return (
    <div className="detail-container">
      <h1>{title}</h1>
      <img src={imageUrl} alt={title} />
      <p>{description}</p>
      {
        favorites.includes(Number(id)) ? <button type="button" disabled> it is in your favorites</button>
          : <button type="button" onClick={handleFavorite}>Mark as favorite</button>
      }
    </div>
  );
};

AnimeDetails.propTypes = {
  match: PropTypes.instanceOf(Object).isRequired,
  animeDetails: PropTypes.instanceOf(Object).isRequired,
  setFavoriteAnime: PropTypes.func.isRequired,
  user: PropTypes.instanceOf(Object).isRequired,
  favorites: PropTypes.instanceOf(Array).isRequired,
  fetchAnimeDetail: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  user: state.user,
  favorites: state.favorites,
  animeDetails: state.animeDetails,
});

export default connect(mapStateToProps, { fetchAnimeDetail, setFavoriteAnime })(AnimeDetails);
