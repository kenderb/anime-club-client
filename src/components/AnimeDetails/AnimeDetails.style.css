.detail-container h1 {
  justify-content: space-between;
  margin: 1em;
  font-size: 1em;
  text-align: center;
}

.detail-container img {
  object-fit: cover;
  height: 20em;
}

.detail-container p {
  padding: 1em;
}

.favorite-button-base {
  width: 100%;
  padding: 1em;
  color: white;
  border: none;
  background-color: var(--primary-color);
}

.favorite-desable {
  color: #e6e6e6;
  background-color: #8383835d;
}
