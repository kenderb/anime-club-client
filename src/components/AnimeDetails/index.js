import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link, Redirect } from 'react-router-dom';
import { ChevronBackOutline, SearchOutline } from 'react-ionicons';
import { fetchAnimeDetail, setFavoriteAnime } from '../../actions';
import './AnimeDetails.style.css';

const AnimeDetails = ({
  match, animeDetails, user, setFavoriteAnime, favorites,
  fetchAnimeDetail,
}) => {
  const { id } = match.params;
  const { title, url_image: imageUrl, description } = animeDetails;

  const handleFavorite = () => {
    setFavoriteAnime({ userId: user.id, animeId: id });
  };

  useEffect(() => {
    fetchAnimeDetail(id);
  }, [fetchAnimeDetail]);

  if (!user.loggedIn) return <Redirect to="/" />;

  return (
    <div className="detail-container">
      <h1 className="d-flex">
        <Link to="/favorites">
          <ChevronBackOutline
            color="#00000"
            height="25px"
            width="25px"
          />
        </Link>
        {title}
        <SearchOutline
          color="#00000"
          height="25px"
          width="25px"
        />
      </h1>
      <img src={imageUrl} alt={title} />
      <p>{description}</p>
      {
        favorites.includes(Number(id)) ? '' : <button type="button" onClick={handleFavorite} className="favorite-button-base">Mark as favorite</button>
      }
    </div>
  );
};

AnimeDetails.propTypes = {
  match: PropTypes.instanceOf(Object).isRequired,
  animeDetails: PropTypes.instanceOf(Object).isRequired,
  setFavoriteAnime: PropTypes.func.isRequired,
  user: PropTypes.instanceOf(Object).isRequired,
  favorites: PropTypes.instanceOf(Array).isRequired,
  fetchAnimeDetail: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  user: state.user,
  favorites: state.favorites,
  animeDetails: state.animeDetails,
});

export default connect(mapStateToProps, { fetchAnimeDetail, setFavoriteAnime })(AnimeDetails);
