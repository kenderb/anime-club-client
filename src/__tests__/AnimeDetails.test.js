import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { createStore, applyMiddleware } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import combineReducers from '../reducers';
import AnimeDetails from '../components/AnimeDetails';

test('AnimeDetails renders correctly', () => {
  const mockObj = { params: 'some params' };
  const store = createStore(combineReducers, applyMiddleware(thunk));
  const tree = create(
    <Provider store={store}>
      <BrowserRouter>
        <AnimeDetails match={mockObj} />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
