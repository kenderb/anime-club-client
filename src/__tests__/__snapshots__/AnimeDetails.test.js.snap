// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimeDetails renders correctly 1`] = `null`;
